CONTRIBUTING.md

```diff
@@ -0,0 +1,124 @@
+<img src="assets/amber.png" alt="amber logo" width="250" align="right" />
+
+# Contributing to Amber
+This is a simple but exhaustive guide to get you started on contributing to amber.
+
+## Contributing guidelines
+Before you dig into Amber, you should know a few things before you contribute.
+
+Any code change is submitted [through a PR](https://github.com/Ph0enixKM/Amber/pulls), which is then approved by at least 2 maintainers.
+
+The way we talk on github is not the same as we would talk in person. When on github, always get straight to the point and be critical.
+
+Personal grudges are forbidden around here, as well as anything offtopic or offensive.
+
+### Opening a PR
+
+Before a PR is opened, it usually has an issue about it first, where we discuss how exactly a feature must be implemented, to avoid making a mistake.
+
+It is recommended that you see how features were already implemented. A good example is [#130](https://github.com/Ph0enixKM/Amber/issues/130)
+
+To create a PR, you should fork the repo, create a branch, do your work in there, and open a PR. It will then be reviewed and pushed into master.
+
+### Getting help
+On the way you will need to get some help with your code. The best way to ask is in [our Discord server](https://discord.com/invite/cjHjxbsDvZ), but you may also ask other contributors personally or post in [Discussions](https://github.com/Ph0enixKM/Amber/discussions).
+
+## Overview
+Amber consists of the following layers:
+
+1. [CLI Interface](#1-cli-interface)
+2. [Compiler](#2-compiler)  
+   1. [Parser & tokenizer](#21-parser--tokenizer)
+   2. [Translator](#22-translator)
+3. [Runtime libraries](#3-runtime-libraries)
+   1. [`stdlib`](#31-stdlib)
+4. [Tests](#4-tests)
+
+### 1. CLI Interface
+All CLI interface is in [`main.rs`](src/main.rs). [`clap`](https://crates.io/crates/clap) handles argument parsing.
+
+### 2. Compiler
+Compiler consists of these files: [`compiler.rs`](src/compiler.rs), and everything in [`modules`](src/modules).
+
+`AmberCompiler` struct by itself is just a bootstrapper for all the syntax modules.
+
+#### 2.1. Parser & tokenizer
+Thanks to [`heraclitus`](https://github.com/Ph0enixKM/Heraclitus), we can use simple abstractions to go through tokens.
+
+Please open any syntax module code file, and find a line that says: `impl SyntaxModule<ParserMetadata> for MODULE_NAME_HERE`
+
+It will have a `parse()` function, where all the magic happens. You can either dig into the code yourself or look at the example below to understand how it works.
+
+<details>
+<summary>Example parser</summary>
+
+**Important: this is pseudo code. Its purpose is to demonstrate how it should look like.**
+
+```rs
+// This code parses the following: `1 + 2`
+fn parse(meta: ParserMetadata) {
```

Comment:
We take a `& mut` so that we don't have to copy the metadata around. Also the function returns `SyntaxResult` and that's good to explicitly declare to the reader.

```suggestion
fn parse(meta: &mut ParserMetadata) -> SyntaxResult {
```